zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -1145,3 +1145,17 @@ def test_910_restart_on_config_change(self):
             {}, {},
             self.services)
         logging.info("Passed restart on changed test.")
+
+    def test_action_bootstrap_mysqlrouter(self):
+        """Checking bootstrap-mysqlrouter action.
+
+        Run the bootstrap-mysqlrouter action
+        """
+        logging.info("Running bootstrap-mysqlrouter action")
+        action = zaza.model.run_action_on_leader(
+            self.application,
+            "bootstrap-mysqlrouter",
+            action_params={})
+        assert "Success" in action.data["results"]["outcome"], (
+            "Bootstrap mysqlrouter action failed.")
+        logging.info("Passed bootstrap-mysqlrouter action test.")
```

Comment:
This is a good start.  It is a "happy path" test to just verify that the bootstrap action works.

A better test will be to break the mysql-router instance (e.g. overwrite a critical file, ensure that the router is broken), and then run the action and then verify that the mysql-router is actually working.  i.e. this test doesn't actually verify that the action works.

Reply:
Thanks I'm going to do this in  this week.